### DISCUSSION TITLE
tutorials/federation-hotchocolate/01-intro-federation

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#183] [course feedback]
# tutorials/federation-hotchocolate/01-intro-federation

https://www.apollographql.com/tutorials/federation-hotchocolate/01-intro-federation

**Comment by scottrudy** [comment]
In today's world are we really suggesting that we blindly run a shell script to install anything? Horrible idea

**Reply by scottrudy** [reply]
It would be much better if these applications could be installed from an OS appropriate package manager such as APT, HOMBREW or WINGET.

**Reply by lizhennessy** [reply]
Hi there @scottrudy , are you referring to the Rover installation command, or the commands to run (install & build) the project?

**Reply by scottrudy** [reply]
The Rover utility. The current recommendation is to navigate to a url, download a shell script and blindly execute it.

**Reply by lizhennessy** [reply]
Ah, gotcha. We provide the `curl` script because it's a quick and easy way of getting the Rover binary installed on your machine, but I understand if that feels problematic. We've also provided a [link to the documentation](https://www.apollographql.com/docs/rover/getting-started#installation-methods) which details alternative ways of installing Rover, including adding it as a dependency in your project.

**Reply by jernejg** [reply]
> It would be much better if these applications could be installed from an OS appropriate package manager such as APT, HOMBREW or WINGET.

or a docker container :)
